dclab/correct.py

```diff
@@ -0,0 +1,49 @@
+"""Functions used to correct RTDC-files"""
+
+import pathlib
+import shutil
+
+import h5py
+import numpy as np
+
+
+def offset(path_in, path_out):
+    """Computes trace offset and corrects features "fl?_max" accordingly
+
+    Parameters
+    ----------
+    path_in: str or pathlib.Path
+        Path of input file for which the offset will be corrected.
+    path_out: str or pathlib.Path
+        Path where file with corrected offset will be saved
+    """
+    path_in = pathlib.Path(path_in)
+    path_out = pathlib.Path(path_out)
+
+    if path_out.suffix != ".rtdc":
+        path_out = path_out.with_name(path_out.name + ".rtdc")
+
+    if path_out.exists():
+        raise ValueError("Output file '{}' already exists!".format(path_out))
+
+    shutil.copy2(path_in, path_out)
+
+    with h5py.File(path_out, 'r+') as hf:
+        ds = hf["events"]
+
+        for ii in range(1, 4):
+            # skip if offset already corrected
+            feat = "fl{}_raw".format(ii)
+            feat_max = "fl{}_max".format(ii)
+            baseline_str = "fluorescence:baseline {} offset".format(ii)
+
+            if baseline_str in hf.attrs:
+                continue
+
+            if feat in ds["trace"] and feat_max in ds:
+                trace = ds["trace"][feat]
+                offset = np.min(trace)
+                # Only consider negative offsets
+                offset = min(offset, 1)
+                hf.attrs["fluorescence:baseline {} offset".format(ii)] = offset
+                ds["fl{}_max".format(ii)][:] -= (offset-1)
```

Comment:
Offset correction should only happen for datasets that have negative values in the trace. For datasets where offset is "0", nothing should happen. Please add that case for clarity here and add a test function. "fluorescence:baseline {} offset" should be written in any case.

Reply:
Removing the `1`s here and in the code you mention in the comment above, should resolve this issue, right?
We then would have `ds["fl{}_max".format(ii)][:] -= offset`, which would make it (implicitly) clear that no adjustment is made, if offset is 0. Should I make this clearer in the code anyways?